core/components/commerce_timeslots/src/Admin/Schedule/Form.php

```diff
@@ -2,31 +2,104 @@
 
 namespace modmore\Commerce_TimeSlots\Admin\Schedule;
 
-use comOrderAddress;
+use modmore\Commerce\Admin\Widgets\Form\SectionField;
+use modmore\Commerce\Admin\Widgets\Form\SelectMultipleField;
 use modmore\Commerce\Admin\Widgets\Form\TextField;
 use modmore\Commerce\Admin\Widgets\FormWidget;
+use modmore\Commerce_TimeSlots\Modules\TimeSlots;
 
 /**
- * Class Form
- * @package modmore\Commerce\Admin\Configuration\PaymentMethods
- *
- * @property comOrderAddress $record
+
+ * @property \ctsSchedule $record
  */
 class Form extends FormWidget
 {
     protected $classKey = 'ctsSchedule';
     public $key = 'timeslots-schedule-form';
     public $title = '';
+    public $availableDays = [];
 
     public function getFields(array $options = array())
     {
+        // Create structure of day options
+        $this->availableDays = [
+            [
+                'label' => $this->adapter->lexicon('sunday'),
+                'value' => TimeSlots::SUNDAY
+            ],
+            [
+                'label' => $this->adapter->lexicon('monday'),
+                'value' => TimeSlots::MONDAY
+            ],
+            [
+                'label' => $this->adapter->lexicon('tuesday'),
+                'value' => TimeSlots::TUESDAY
+            ],
+            [
+                'label' => $this->adapter->lexicon('wednesday'),
+                'value' => TimeSlots::WEDNESDAY
+            ],
+            [
+                'label' => $this->adapter->lexicon('thursday'),
+                'value' => TimeSlots::THURSDAY
+            ],
+            [
+                'label' => $this->adapter->lexicon('friday'),
+                'value' => TimeSlots::FRIDAY
+            ],
+            [
+                'label' => $this->adapter->lexicon('saturday'),
+                'value' => TimeSlots::SATURDAY
+            ],
+        ];
+
         $fields = [];
 
         $fields[] = new TextField($this->commerce, [
             'name' => 'name',
             'label' => $this->adapter->lexicon('commerce.name'),
         ]);
 
+        // Requires the Scheduler service in Commerce 1.3+
+        $root = $this->commerce->config['core_path'];
+        $path = $root . 'src/Services/Scheduler/crontime';
+        if (file_exists($path)) {
```

Comment:
I'd suggest moving this logic to a `$this->commerce->isSchedulerActive()` method (perhaps with a `method_exists()` to prevent errors pre-1.3)... modules, even our own, shouldn't really have to resort to checks like this.

Perhaps also worthwhile changing it to `if (not active) { show warning that it requires the scheduler }` and still allow setting up the default schedules before 1.3/scheduler is active?